akka-http-caching/src/main/scala/akka/http/caching/LfuCache.scala

```diff
@@ -0,0 +1,91 @@
+package akka.http.caching
+
+import java.util.concurrent.{ CompletableFuture, Executor, TimeUnit }
+
+import scala.collection.JavaConverters._
+import scala.concurrent.duration.Duration
+import scala.concurrent.Future
+import com.github.benmanes.caffeine.cache.{ AsyncCacheLoader, AsyncLoadingCache, Caffeine }
+import akka.http.caching.LfuCache.{ toJavaMappingFunction, dummyLoader }
+
+import scala.compat.java8.FutureConverters._
+import scala.compat.java8.FunctionConverters._
+
+object LfuCache {
+
+  /**
+   * Creates a new [[akka.http.caching.ExpiringLfuCache]] or
+   * [[akka.http.caching.SimpleLfuCache]] instance depending on whether
+   * a non-zero and finite timeToLive and/or timeToIdle is set or not.
+   */
+  def apply[V](
+    maxCapacity:     Int      = 500,
```

Comment:
In the simplest case it would just be a case class that carries all the arguments. However, because case classes are likely to break binary compatibility when changes are necessary we now usually use a public abstract class / private case class implementation combo. Basically like this:

```scala
abstract class LfuCacheSettings private[http]() { self: LfuCacheSettingsImpl =>
 def maxCapacity:     Int
 def initialCapacity: Int
 def timeToLive:      Duration
 def timeToIdle:      Duration 

  def withMaxCapacity(newMaxCapacity: Int): LfuCacheSettings = copy(maxCapacity = newMaxCapacity)
 // similar for other settings
}
object LfuCacheSettings {
  def apply(): LfuCacheSettings = LfuCacheSettingsImpl()
  
  private[http] case class LfuCacheSettingsImpl(
      maxCapacity:     Int      = 500,
      initialCapacity: Int      = 16,
      timeToLive:      Duration = Duration.Inf,
      timeToIdle:      Duration = Duration.Inf
  ) extends LfuCacheSettings
}
```

then the signature could be

```scala
def apply[V](settings: LfuCacheSettings = LfuCacheSettings()): Cache[V]
```

It's a bit of boilerplate but it is helpful later when changes might become necessary.